src/shims/sync.rs

```diff
@@ -58,8 +58,31 @@ fn set_at_offset<'mir, 'tcx: 'mir>(
 // store an i32 in the first four bytes equal to the corresponding libc mutex kind constant
 // (e.g. PTHREAD_MUTEX_NORMAL).
 
+/// A flag that allows to distinguish `PTHREAD_MUTEX_NORMAL` from
+/// `PTHREAD_MUTEX_DEFAULT`. Since in `glibc` they have the same numeric values,
+/// but different behaviour, we need a way to distinguish them. We do this by
+/// setting this bit flag to the `PTHREAD_MUTEX_NORMAL` mutexes. See the comment
+/// in `pthread_mutexattr_settype` function.
+const PTHREAD_MUTEX_NORMAL_FLAG: i32 = 0x8000000;
```

Comment:
We should probably assert that this... doesn't somehow collide badly with the `PTHREAD_MUTEX_*` constants?

Reply:
Oh, you do that already. Never mind :)